### PR TITLE
utils/chunker: fix blocking done chan. fixes #467

### LINKED ISSUE #467: Chunker test log reader hangs up randomly
```chunker $ go test -v
=== RUN   TestNew
--- PASS: TestNew (0.00s)
=== RUN   TestLogReader
^Csignal: interrupt
FAIL	github.com/mesg-foundation/core/utils/chunker	2.477s
chunker $ go test -v
=== RUN   TestNew
--- PASS: TestNew (0.00s)
=== RUN   TestLogReader
^Csignal: interrupt
FAIL	github.com/mesg-foundation/core/utils/chunker	2.030s
chunker $ go test -v
=== RUN   TestNew
--- PASS: TestNew (0.00s)
=== RUN   TestLogReader
--- PASS: TestLogReader (0.00s)
PASS
ok  	github.com/mesg-foundation/core/utils/chunker	0.009s```

### PR DESCRIPTION
**Pull request by ilgooz**

### CODE REVIEW

**Comment by krhubert** [comment]
In that case, `done` channel is not closed anywhere and it leads to leaks.

File: utils/chunker/stream.go [hunk 1]
```diff
@@ -49,10 +54,12 @@ func (s *Stream) Read(p []byte) (n int, err error) {
 
 // Close closes reader.
 func (s *Stream) Close() error {
+	s.m.Lock()
+	defer s.m.Unlock()
 	if s.done == nil {
 		return nil
 	}
-	close(s.done)
```
Comment (krhubert) [inline utils/chunker/stream.go]: In that case, done channel is not closed anywhere and it leads to leaks.

Comment (ilgooz) [reply utils/chunker/stream.go]: it is garbage collected. non-closed chans does not cause memory leaks

Comment (krhubert) [reply utils/chunker/stream.go]: True, ok for me then

**Pull request merged**